### ISSUE TITLE
DOMException: Failed to execute 'getSubStringLength' on 'SVGTextContentElement': The charnum provided (0) is greater than or equal to the maximum bound (0)

### ISSUE THREAD

**Issue by valh1996** [apexcharts/apexcharts.js#624]
Hello,

After several hours of searching, I can't find the source of the problem. I'm sorry I didn't make a codepen, but I couldn't reproduce it on it because my architecture is complicated and my data is on my side. I will therefore try to be as clear as possible.

This is a warning and not an error :

```
Possible Unhandled Promise Rejection: DOMException: Failed to execute 'getSubStringLength' on 'SVGTextContentElement': The charnum provided (0) is greater than or equal to the maximum bound (0).
    at Graphics.placeTextWithEllipsis (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:1735:21)
    at XAxis.xAxisLabelCorrections (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:10903:22)
    at Grid.drawGrid (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:15226:17)
    at eval (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:27613:19)
    at doResolve (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:14942:5)
    at new Promise$1 (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:14846:3)
    at ApexCharts.mount (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:27601:14)
    at eval (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:27954:16)
    at doResolve (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:14942:5)
    at new Promise$1 (webpack-internal:///./node_modules/apexcharts/dist/apexcharts.common.js:14846:3)
```

To sum up, my graph (line) loads correctly the first time, no worries to report. I load it via a function that uses a promise and inside the promise I set the series and categories as follows:
```
    .then(res => {
        let data = res.data.data;
        this.processor.series = [
            {
                 data: data.series["user"]
            },
            {
                  data: data.series["nice"]
            },
            {
                  data: data.series["system"]
            },
            {
                  data: data.series["idle"]
            },
        ];

        this.processor.chartOptions.xaxis.categories = data.categories;
     })
```

I declare my graph as follows:
```
<apexchart
    v-else
    type="line"
    height="400"
    :options="processor.chartOptions"
    :series="processor.series"
></apexchart>
```

To finish, I have a datetimepicker that allows me to select two dates (`start` and `end`) to display the graph according to these dates. I know when a date has been selected using a watcher with `VueJS`.

_**It is inside this watcher that I call exactly the same promise as the one when loading the page by passing the new date.**_
```
watch: {
  period: {
    handler(val) {
      this.period.start = val.start;
      this.period.end = val.end;

      // Reload statistics
      if (!this.processor.chartLoading) {
        this.getCpuOverallStatsByPeriod(
          this.$route.params.id,
          moment(this.period.start, "YYYY-MM-DD HH:mm:ss").unix(),
          moment(this.period.end, "YYYY-MM-DD HH:mm:ss").unix()
        );
      }
    },
    deep: true
  }
}
```

I confirm that the data returned by my API and `console.log` into the promise are corrects (an array of 1257).  So I have 4 series of this size and the categories array also.

**In conclusion, the display is updated and there is no visual problem to notice, but this warning appears en masse in the console.** 

I hope I didn't take too long, thank you for your reader.

Best regards,
Valentin

**Comment by junedchhipa** [comment]
It would be great if you could simplify the problem and try to create a reproducible codepen with all complexities removed.
If you're using Vue, you can create a codesandbox demo with this Vue template - https://codesandbox.io/s/pwwz8009n0

Otherwise, it's just difficult to do a guess and get to the root of the issue with your current details.

**Comment by valh1996** [comment]
I couldn't reproduce the problem. But here's the idea: https://codesandbox.io/s/vue-basic-example-ztrwu

Except that in reality the watcher is placed on my date component which is modifies a Vuex state.
The promise calls an API route via axios, but the operation is the same.

**Comment by junedchhipa** [comment]
Ok, I have fixed the original issue in the above commit.
The patch will be released in 3.7.2.
